### PR TITLE
[SPARK-4984][CORE][WEBUI] Adding a pop-up containing the full job description when it is very long

### PR DESCRIPTION
**Pull request by scwf**
In some case the job description will be very long, such as a long sql. refer to #3718
This PR add a pop-up for job description when it is long. 

image

image

### CODE REVIEW

**Comment by scwf** [comment]
Ping @JoshRosen, any comment here?

**Comment by JoshRosen** [comment]
What do you think about using a styled read-only text form field?  Using CSS, we could probably limit the size of what's displayed while still allowing users to double-click to copy the entire field.

**Comment by scwf** [comment]
Using a read-only text form, the screenshot is like this, also support to copy the description.
image

**Comment by scwf** [comment]
@JoshRosen, any comment on the latest version?

**Comment by JoshRosen** [comment]
This looks pretty nice; I like how copy-paste still works.  One minor thought, though: is there any easy way to make it more visually obvious that the description has been truncated?  Maybe we can fade out the right-hand side of the cell or something, or use CSS + Javascript to add  a `...` or something?  Right now, I'm worried that there could be cases where it's hard to spot that it's been truncated:

image

**Comment by scwf** [comment]
Hmm, agree with you, but have not found a easy way to spot truncated description.
if we add `...` for truncated decs, we will consider the case of window scaling and different screen which make this more complex

**Comment by scwf** [comment]
Updated @JoshRosen, now use `...` to spot it's been truncated
image


image

**Comment by JoshRosen** [comment]
Ooh, very nice!  I didn't know about `text-overflow: ellipsis`.

**Comment by JoshRosen** [comment]
I just tried this out in Firefox and it turns out that it won't let you select text from a disabled input element.  Apparently this has been a longstanding issue; this bug has been open for 11+ years: https://bugzilla.mozilla.org/show_bug.cgi?id=195361

**Comment by JoshRosen** [comment]
I think I found a solution that works in Firefox: replace the `input` element with a `span` where you've set it to `display: block` and `white-space: nowrap`.  If we do that, I think we can drop the border, margin, padding, and cursor, and box shadow attributes.

File: core/src/main/resources/org/apache/spark/ui/static/webui.css [hunk 1]
```diff
@@ -120,6 +120,19 @@ pre {
   border: none;
 }
 
+.description-input {
+  overflow: hidden !important;
```
Comment (pwendell) [inline core/src/main/resources/org/apache/spark/ui/static/webui.css]: Why do these all need "!important"? Are there other rules that will take precedence here?

Comment (scwf) [inline core/src/main/resources/org/apache/spark/ui/static/webui.css]: yes, in my debug there is other style will take precedence here. anyway, new version i do not use input and remove "!important"

**Comment by scwf** [comment]
@JoshRosen, yeah, get it, updated.

**Comment by scwf** [comment]
Ping @JoshRosen is this ok？

**Comment by JoshRosen** [comment]
Yep, looks good to me.  Mind updating the screenshot in the PR description?  Do that, ping me, and I'll merge it right away.  Thanks!

**Comment by scwf** [comment]
Ping @JoshRosen updated done

**Comment by JoshRosen** [comment]
Awesome, thanks!  I'm merging this into `master` (1.3.0).

**Comment by JoshRosen** [comment]
I think that this feature may no longer be working in recent versions of Spark.  Have you tested this recently?

**Comment by scwf** [comment]
i tested it locally today, seems it is ok.

**Pull request closed without merging**